src/canvas.class.js

```diff
@@ -418,6 +425,11 @@
         this.clearContext(this.contextTop);
         this.contextTopDirty = false;
       }
+      // while erasing the brush is in charge of rendering the canvas so we return
+      if (this.isErasing()) {
+        this.freeDrawingBrush._render();
+        return;
+      }
```

Comment:
i do not think we should have this. renderAll has the duty to render, and should not know about the actual brush. Devs may want to call it at some point and in case, is renerTopLayer that should refresh the brush.

Reply:
I agree. However, eraser rendering logic hijacks the canvas to render in layers to achieve an erased effect.
I can move it to the mixin logic. What do you think?

Reply:
Moved logic to mixin